### PR TITLE
Add solution and tests for Task5

### PR DESCRIPTION
**Pull request by VikSil**

### CODE REVIEW

File: __tests__/server.test.js [hunk 1]
```diff
@@ -91,48 +91,112 @@ describe("server --> nc_news_test", () =>{
...
-            expect(typeof article.article_id).toBe("number");
-            expect(typeof article.votes).toBe("number");
-            expect(typeof article.title).toBe("string");
-            expect(typeof article.topic).toBe("string");
-            expect(typeof article.author).toBe("string");
-            expect(typeof article.body).toBe("string");
-            expect(typeof article.created_at).toBe("string");
-            expect(typeof article.article_img_url).toBe("string");
+            expect(body.articles).toBeInstanceOf(Array);
+            expect(body.articles.length).toBeGreaterThanOrEqual(1);
```
Comment (nc-beth-mccormack) [inline __tests__/server.test.js:113]: You can be more specific here - how many articles do you expect to get back?

File: __tests__/server.test.js [hunk 2]
```diff
@@ -91,48 +91,112 @@ describe("server --> nc_news_test", () =>{
...
+              expect(article).toBeInstanceOf(Object);
+              expect(Object.keys(article)).toHaveLength(8);
+              expect(typeof article.author).toBe("string");
+              expect(typeof article.title).toBe("string");
+              expect(typeof article.article_id).toBe("number");
+              expect(typeof article.topic).toBe("string");
+              expect(typeof article.created_at).toBe("string");
+              expect(typeof article.votes).toBe("number");
+              expect(typeof article.article_img_url).toBe("string");
+              expect(typeof article.comment_count).toBe("number");
```
Comment (nc-beth-mccormack) [inline __tests__/server.test.js:127]: You can make this test more robust by using .toHaveProperty or .toMatchObject. expect.any could also be useful. Just testing the type of the value is not a great way to check these properties exist

File: __tests__/server.test.js [hunk 3]
```diff
@@ -91,48 +91,112 @@ describe("server --> nc_news_test", () =>{
...
+            });
           });
       });
-      test("Returns status code 200 and correctly formatted body to a correct requests", () => {
+      describe("GET /api/articles/sort_by=...", () => {
```
Comment (nc-beth-mccormack) [inline __tests__/server.test.js:131]: This ticket only asks for created_at to be sorted by descending, by default

Comment (nc-beth-mccormack) [reply __tests__/server.test.js:131]: I would remove these tests until they are necessary, and add a test to ensure the default sort order is correct

**Review by nc-beth-mccormack — COMMENTED** [review]
I've left a few comments 🙂

File: __tests__/server.test.js (hunk 2 above)
Comment (VikSil) [reply __tests__/server.test.js:127]: Thanks so much for this @nc-beth-mccormack , very helpful feedback.

**Pull request merged**